tests/integration/seccomp.bats

```diff
@@ -79,18 +75,68 @@ function teardown() {
 				"syscalls":[{"names":["mkdir"], "action":"SCMP_ACT_ERRNO"}]
 			}'
 
-	declare -A FLAGS=(
-		['REMOVE']=4 # No setting, use built-in default.
-		['EMPTY']=0  # Empty set of flags.
-		['"SECCOMP_FILTER_FLAG_LOG"']=2
-		['"SECCOMP_FILTER_FLAG_SPEC_ALLOW"']=4
-		['"SECCOMP_FILTER_FLAG_TSYNC"']=0 # tsync flag is ignored.
-		['"SECCOMP_FILTER_FLAG_LOG","SECCOMP_FILTER_FLAG_SPEC_ALLOW"']=6
-		['"SECCOMP_FILTER_FLAG_LOG","SECCOMP_FILTER_FLAG_TSYNC"']=2
-		['"SECCOMP_FILTER_FLAG_SPEC_ALLOW","SECCOMP_FILTER_FLAG_TSYNC"']=4
-		['"SECCOMP_FILTER_FLAG_LOG","SECCOMP_FILTER_FLAG_SPEC_ALLOW","SECCOMP_FILTER_FLAG_TSYNC"']=6
+	# This is a set of all possible flag combinations to test.
+	declare -A TEST_CASES=(
+		['EMPTY']=0 # Special value: empty set of flags.
+	)
+
+	# List of flags to test and their respective numeric values.
+	declare -A flags=(
+		["TSYNC"]=0 # Supported but ignored by runc, thus 0.
+		["LOG"]=2
+		["SPEC_ALLOW"]=4
+		# XXX: add new seccomp flags above this line.
 	)
-	for key in "${!FLAGS[@]}"; do
+
+	# Get the list of flags supported by runc/seccomp/kernel.
+	mapfile -t sup_flags < <(__runc features |
+		jq '.linux.seccomp.supported_flags' | grep 'SECCOMP' |
+		sed -e 's/^ *"SECCOMP_FILTER_FLAG_//' -e 's/",*$//')
+
+	# Check that any new seccomp flags added are actually tested.
+	for f in "${sup_flags[@]}"; do
+		if ! [ -v 'flags[$f]' ]; then
+			echo "ERROR: seccomp flag $f not added to the test; fix the text!" 1>&2
+			exit 1
+		fi
+	done
+
+	# Filter out flags not supported by the current seccomp/kernel,
+	# so this test can be run on older systems.
```

Comment:
I'm sure I'm missing something, but why hardcode the vars in the `flags` array and then ignore the not supported ones? Wouldn't be the same to just test all the supported ones? (and we don't harcode any flag here)

I see we might get unlucky in the CI and maybe some flag is never tested, and that is interesting to know. But we are not detecting that here either, right? Maybe that can be a different test for that (well, not sure it is easy to detect that... and seems unlikely)?

Reply:
Since this test is actually checking that the flags are set as expected (the check was added by commit 26dc55ef1a56ea02794 after we found out in #3580 that SPEC_ALLOW is just ignored), we need to know their numeric values.

Surely, we can just get the list of supported flags (which we should test) from `runc features`, but then again we still need their values.

Let me see if I can refactor the code to simplify it more (use supported flags as received from runc directly, and only use the hardcoded ones for numeric values).

Reply:
Oh, and wouldn't make sense for `runc features` to also print the numeric values (if we can easily parse it here, not worth the pain if this is hard)?

But the code is okay as it is now too from my POV :)

Reply:
> Oh, and wouldn't make sense for runc features to also print the numeric values (if we can easily parse it here, not worth the pain if this is hard)?

Well, no one ever needs the numeric values (except for this test), and `runc features` target audience is runc users, not our tests.

Thus we have to hardcode it (unfortunately there is no easy way to see if a seccomp flag is set or not, so the test have to rely on debug output and compare the numbers).

Reply:
Ohh, ok. I thought it might have been present on some errors as hex or something maybe. But makes sense, in any case it is not so useful either